subsys/net/lib/mqtt_sn/mqtt_sn_transport_udp.c

```diff
@@ -43,25 +44,65 @@ static int tp_udp_init(struct mqtt_sn_transport *transport)
 {
 	struct mqtt_sn_transport_udp *udp = UDP_TRANSPORT(transport);
 	int err;
+	struct sockaddr_in addrm;
+	struct ip_mreqn mreqn;
+	int optval;
+	struct net_if *iface;
 
-	udp->sock = zsock_socket(udp->gwaddr.sa_family, SOCK_DGRAM, 0);
+	udp->sock = zsock_socket(udp->bcaddr.sa_family, SOCK_DGRAM, 0);
 	if (udp->sock < 0) {
 		return errno;
 	}
 
 	LOG_DBG("Socket %d", udp->sock);
 
+	/* Enable broadcasting mode */
+	/*
+	 * int broadcastEnable = 1;
+	 * err = zsock_setsockopt(udp->sock, SOL_SOCKET, SO_BROADCAST,
+	 *						 &broadcastEnable, sizeof(broadcastEnable));
+	 * if (udp->sock < 0) {
+	 *	return errno;
+	 * }
+	 */
+
+	optval = 1;
+	err = zsock_setsockopt(udp->sock, SOL_SOCKET, SO_REUSEADDR, &optval, sizeof(optval));
+	if (err < 0) {
+		return errno;
+	}
+
 #ifdef LOG_DBG
 	char ip[30], *out;
 
-	out = get_ip_str((struct sockaddr *)&udp->gwaddr, ip, sizeof(ip));
+	out = get_ip_str((struct sockaddr *)&udp->bcaddr, ip, sizeof(ip));
 	if (out != NULL) {
-		LOG_DBG("Connecting to IP %s:%u", out,
-			ntohs(((struct sockaddr_in *)&udp->gwaddr)->sin_port));
+		LOG_DBG("Binding to Brodcast IP %s:%u", out,
+			ntohs(((struct sockaddr_in *)&udp->bcaddr)->sin_port));
 	}
 #endif
```

Comment:
My comment is not directly related to your changes. It would actually be better to use the actual debug enabling option here.
So something like
```
if (IS_ENABLED(CONFIG_MQTT_SN_LOG_LEVEL_DBG)) {
....
}
```
This way the debugging code will be compile tested but will not be enabled unless the debugging option is enabled.